### ISSUE TITLE
Implement AWS AddUserToGroup as top-level resource

### ISSUE THREAD

**Issue by GrayCoder** [hashicorp/terraform#2274]
Currently, [this functionality](https://docs.aws.amazon.com/IAM/latest/APIReference/API_AddUserToGroup.html) isn't supported by terraform and I'd like to figure out the best way to implement it. @ctiwald and I need this functionality in order to port our current configuration to terraform tf files.

It would seem that, based on issues like #2197 and #1419, it's best to implement this as a top level resource.

Proposed format:
```
resource "aws_iam_group_association" "dev" {
    name = "dev"
    users = [
        "${aws_iam_user.dev1.name}",
        "${aws_iam_user.dev2.name}",
        ...
    ]
    group = "${aws_iam_group.dev.name}"
}
```

with files:
`builtin/providers/aws/resource_aws_iam_group_association.go`
`builtin/providers/aws/resource_aws_iam_group_association_test.go`

Does this sound acceptable or is there another preferred method/syntax?

**Comment by catsby** [comment]
Hey @GrayCoder – I opened https://github.com/hashicorp/terraform/pull/2273 a few hours ago for this :smile: 

It's currently just a 1-to-1 mapping of user to group though, mostly just to get the ball rolling. Does anyone mentioned here think strongly enough that it needs to be 1-to-many?

**Comment by catsby** [comment]
Implemented in #2273
